digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/ProjectLinkSplitter.scala

```diff
@@ -135,35 +135,42 @@ object ProjectLinkSplitter {
   }
 
   def split(suravage: ProjectLink, templateLink: ProjectLink, splitOptions: SplitOptions): Seq[ProjectLink] =
-    split(suravage, templateLink, Seq(), splitOptions).toSeqWithAllTerminated
+    split(suravage, templateLink, splitOptions)
```

Comment:
Here we have an infinite recursion, which can be seen when running ProjectLinkSplitterSpec: "Split aligning project links".

Reply:
I've added one fix for the older split test, and also added one new when there are 2 terminated links with same linkId. @Hugo can check the last one tho so see it needs some more rule/validation